From 2ec4cae2d121234958bdc38074a4f14bf1f6123f Mon Sep 17 00:00:00 2001
From: Guy Nisan <gnisan@habana.ai>
Date: Mon, 21 Sep 2020 13:53:51 +0300
Subject: [PATCH] [SW-13981] gaudi: pci: set id 0x1001 for step b

Set PCI ID 0x1001 for Gaudi step B according to device id read from
eFuse.

Change-Id: I8d9e02784b7774d8c21596bfcaf13a169acf9add
Signed-off-by: Guy Nisan <gnisan@habana.ai>
---

diff --git a/gaudi/common.c b/gaudi/common.c
index b7fd759..9f3996f 100644
--- a/gaudi/common.c
+++ b/gaudi/common.c
@@ -17,6 +17,7 @@
 
 #include <hl_boot_if.h>
 #include <gaudi/gaudi_reg_map.h>
+#include <common/pci_ids.h>
 
 #include <preboot.h>
 #include <pll.h>
@@ -25,6 +26,7 @@
 #include <spi.h>
 #include <preboot/fit.h>
 #include <preboot/thermal.h>
+#include <efuse.h>
 
 #include "common.h"
 
@@ -95,15 +97,35 @@
 	preboot_init_pcie_phy_fw_update();
 }
 
+#define FUSE_DEV_ID_WORD_OFFSET	1
+
+static u32 preboot_get_vendor_id_device_id(void)
+{
+	u32 pci_dev_id = PCI_IDS_GAUDI;
+	char devid[5];
+
+	/* Read the second part of the device ID which contains the step */
+	if (!preboot_fuse_read(FUSE_DEV_ID_WORD_OFFSET + 1, (u32 *)devid)) {
+		devid[4] = '\0';
+		/* Check for device type B1 and set the PCI ID accordingly */
+		if (!strcmp(&devid[2], "B1"))
+			pci_dev_id = PCI_IDS_GAUDI_HL2000M;
+	}
+
+	return (pci_dev_id << 16) | PCI_VENDOR_ID_HABANALABS;
+}
+
 static void preboot_init_pcie(void)
 {
+	u32 ven_id_dev_id = preboot_get_vendor_id_device_id();
+
 	preboot_init_pcie_phy();
 
 	/* Update Vendor ID */
-	regmap_writel(0x10001DA3, mmPCIE_DBI_DEVICE_ID_VENDOR_ID_REG);
+	regmap_writel(ven_id_dev_id, mmPCIE_DBI_DEVICE_ID_VENDOR_ID_REG);
 
 	/* Update subsystem Vendor ID */
-	regmap_writel(0x10001DA3,
+	regmap_writel(ven_id_dev_id,
 		      mmPCIE_DBI_SUBSYSTEM_ID_SUBSYSTEM_VENDOR_ID_REG);
 
 	/* Disable the EQ REDO Gen3 */
